tests/lib/prepare-restore.sh

```diff
@@ -634,6 +634,11 @@ prepare_suite_each() {
         "$TESTSTOOLS"/cleanup-state pre-invariant
     fi
     tests.invariant check
+
+    SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
+    if os.query is-focal || os.query is.fedora; then
```

Comment:
It's not immediately obvious why this should be restricted to focal and fedora. I'd suggest to create a has_inotify_tools function and use it here. Of course, for the time being the function body would check if the OS is focal or fedora, but in the future we can extend it as inotify-tools becomes available on more distros.

Reply:
Alternatively, check if `inotifywait` is available on PATH?

Reply:
Right, my idea was to add this to some systems and then make it available to any system. I'll add the dependencies and make it work on all the supported classic systems. 
Thanks for the review